src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3WebtransportSession.cs

```diff
@@ -0,0 +1,196 @@
+            {
+                return null;
+            }
+        }
+
+        public bool TryGetIncomingWebtransportStreamFromServer(out QuicStream? quicStream)
+        {
+            return _disposed == 1
+                ? throw new ObjectDisposedException(nameof(Http3WebtransportSession))
+                : _incomingStreamsQueue.Reader.TryRead(out quicStream);
+        }
+
+        internal void AcceptServerStream(QuicStream stream)
+        {
+            if (_disposed == 1)
+            {
+                // WebtransportSessionGone error code
+                stream.Abort(QuicAbortDirection.Read, 0x107d7b68);
+                return;
+            }
+
+            bool added = _incomingStreamsQueue.Writer.TryWrite(stream);
+            Debug.Assert(added);
+        }
+
+        internal async Task AbortIncomingSessionWebtransportStreamsAsync(long errorCode, CancellationToken cancellationToken = default)
+        {
+            // check if they were not aborted before
+            if(_incomingStreamsQueue.Writer.TryComplete())
+            {
+                Runtime.CompilerServices.ConfiguredCancelableAsyncEnumerable<QuicStream> incomingStreams = _incomingStreamsQueue.Reader.ReadAllAsync(cancellationToken).ConfigureAwait(false);
+                await foreach (QuicStream incomingStream in incomingStreams)
+                {
+                    incomingStream.Abort(QuicAbortDirection.Read, errorCode);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="QuicStream">quic stream and sends it to the server</see>.
+        /// </summary>
+        public async ValueTask<QuicStream?> OpenWebtransportStreamAsync(QuicStreamType type, CancellationToken cancellationToken = default)
+        {
+            if (_disposed == 1)
+                throw new ObjectDisposedException(nameof(Http3WebtransportSession));
+            return await _WebtransportManager.CreateClientStreamAsync(type, Id, cancellationToken).ConfigureAwait(false);
+        }
+        private void RemoveFromSessionsDictionary()
+        {
+            _WebtransportManager.DeleteSession(Id);
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            if (Interlocked.Exchange(ref _disposed, 1) == 1)
+                return;
+            RemoveFromSessionsDictionary();
+
+            await AbortIncomingSessionWebtransportStreamsAsync((long)0x107d7b68).ConfigureAwait(false);
+            await _connectStream.DisposeAsync().ConfigureAwait(false);
+        }
+
+        public void Dispose()
+        {
+            if (Interlocked.Exchange(ref _disposed, 1) == 1)
+                return;
+            RemoveFromSessionsDictionary();
+            _connectStream.Dispose();
+            _incomingStreamsQueue.Writer.Complete();
+            while (_incomingStreamsQueue.Reader.TryRead(out QuicStream? stream))
+            {
+                stream!.Abort(QuicAbortDirection.Read, 0x107d7b68);
```

Comment:
Fixed it to constant, I do not know why but previously it did not let me use the error

Dispose and dispose Async do the same thing, but instead of creating an async version of `AbortIncomingInboundStreamsAsync` I just added the code below. Should I make a sync function for it? I do not feel like it is a necessity

Reply:
I see it now, thanks.
Maybe you can add a comment like "sync version of AbortIncomingInboundStreamsAsync", so it would be more clear.

I also think you should abort both sides of the incoming streams and dispose them as well, what do you think?

Reply:
added comment and aborted both sides